### DISCUSSION TITLE
Breakdown of First Load JS metric result in build output

### DISCUSSION THREAD

**Discussion by Ziker22** [vercel/next.js#19326] [Help] [answered]
Hi all,
Here is my build output
![image](https://user-images.githubusercontent.com/8571609/99702916-b8dec400-2a96-11eb-9a9b-9250e9dd9891.png)

Now i would expect that "First load JS" column is sum of page size and "Js Shared by all"
At least i got that impression from https://github.com/vercel/next.js/pull/10014
I have also checked doc improvement in https://github.com/vercel/next.js/pull/12034 but im not wiser

Im really wondering how next calculates that number and why its so high
Thx in advance fo any insight

**Comment by Timer** [comment, marked as answer]
The First Load JS size also includes bundles which are not explicitly listed in the output, which includes files shared by **some pages, but not all**.

For example, you may be using `lodash` in only `/product-page` and `/search-page`, which would be 24kb not listed in the page "Size" or in "shared by all".

The "size" column for a page only captures JS **unique to that page**.

**Reply by Ziker22** [reply]
Ah that makes sense thanks 👍

**Reply by kweiberth** [reply]
@Timer I'm using `@auth0/auth0-react` module, which has a `<Provider />` which I'm using to wrap pages in my custom `_app.js`. Unfortunately, this module is huge, like 39kb, and I'm trying to avoid having it be part of the first load JS for some of our public pages. I've set up an if statement in _app and I'm dynamically requiring/rendering this module only for pages that need it. I've confirmed by looking at `build-manifest.json` that this chunk is not included as a chunk for these public pages. However, I'm seeing it listed under "First Load JS shared by all" and I'm seeing it being loaded in the browser for my public pages. 

I see in your response above you mention:

> The First Load JS size also includes bundles which are not explicitly listed in the output, which includes files shared by some pages, but not all.

Does this mean that if a chunk is included in some files, then it gets loaded as "required first load JS" by all pages? Is there a way to change the configuration of this so that this particular chunk is only loaded for pages that require it, and not for all pages?

**Reply by kweiberth** [reply]
I think maybe it's actually not possible to dynamically include a `<Provider />` to different pages in _app.js. I am trying this: 

```js
// _app.js

export default function App({ Component, pageProps }) {
  const router = useRouter()

  if (router.pathname === '/a') {
    return <Component {...pageProps} />
  }

  const { Auth0Provider } = require('@auth0/auth0-react')
  return <Auth0Provider><Component {...pageProps} /></Auth0Provider>
}
```

But I think @auth0/auth0-react is going to get bundled as part of _app.js for all pages. I think I should probably build a wrapper that can be shared around whichever pages need it, but then I think I'll have issues with the provider being unmounted/mounted on each page visit.
